barsys/views.py

```diff
@@ -882,6 +848,57 @@ def get(self, request, user_id):
         return render(request, "barsys/main/user_purchase.html", context)
 
 
+def purchase_no_free_item_single_user_single_purchase_form(form):
```

Comment:
Holy macaroni, we probably shouldn't add any more qualifiers to the method name :D